src/scripts/extensions/tooltipHelper.ts

```diff
@@ -33,22 +35,38 @@ export class TooltipHelper {
 		this.storage.setValue(storageKey, value);
 	}
 
-	public tooltipDelayIsOver(tooltipType: TooltipType, time: number) {
-		if (Utils.isNullOrUndefined(tooltipType) || Utils.isNullOrUndefined(time)) {
+	public tooltipDelayIsOver(tooltipType: TooltipType, curTime: number): boolean {
+		if (Utils.isNullOrUndefined(tooltipType) || Utils.isNullOrUndefined(curTime)) {
 			throw new Error("Invalid argument passed to tooltipDelayIsOver");
 		}
 
+		let numTimesTooltipHasBeenSeen = this.getTooltipInformation(ClipperStorageKeys.numTimesTooltipHasBeenSeenBase, tooltipType);
 		let lastSeenTooltipTime = this.getTooltipInformation(ClipperStorageKeys.lastSeenTooltipTimeBase, tooltipType);
-		let lastSeenClipTime = this.getTooltipInformation(ClipperStorageKeys.lastClippedTooltipTimeBase, tooltipType);
+		let lastClipTime = this.getTooltipInformation(ClipperStorageKeys.lastClippedTooltipTimeBase, tooltipType);
 
-		let timeBetweenTooltips = Constants.Settings.timeBetweenTooltips;
+		// If the user has clipped this content type
+		if (lastClipTime !== 0) {
+			return false;
+		}
+
+		// If the user has seen enough of our tooltips :P 
+		if (numTimesTooltipHasBeenSeen >= Constants.Settings.maximumNumberOfTimesToShowTooltips) {
+			return false;
+		}
 
-		let isTooltipShownDelayOver = (time - lastSeenTooltipTime) >= timeBetweenTooltips;
-		let isClipDelayOver = (time - lastSeenClipTime) >= timeBetweenTooltips;
+		// If it has been less than 3 weeks since the user saw this specific tooltip, return false
+		if (this.tooltipHasBeenSeenInLastTimePeriod(tooltipType, curTime, Constants.Settings.timeBetweenSameTooltip)) {
+			return false;
+		}
 
-		// Only shows if it has been >= 3 weeks since the User saw a tooltip AND it has been >= 3 weeks
-		// since their last Clip of this specific TooltipType
-		return isTooltipShownDelayOver && isClipDelayOver;
+		// If it has been less than 7 days since the user saw any tooltip, then return false
```

Comment:
I think they are readable enough. While its a low level of implementation all the blocks are at the same level.